src/lib/mina.ts

```diff
@@ -328,6 +336,21 @@ function LocalBlockchain({
     },
     addAccount,
     testAccounts,
+    setTimestamp(ms: UInt64) {
+      networkState.timestamp = ms;
+    },
+    setGlobalSlot(slot: UInt32) {
+      networkState.globalSlotSinceGenesis = slot;
+    },
+    setGlobalSlotSinceHardfork(slot: UInt32) {
+      networkState.globalSlotSinceHardFork = slot;
+    },
+    setBlockchainLength(height: UInt32) {
+      networkState.blockchainLength = height;
+    },
+    setTotalCurrency(currency: UInt32) {
```

Comment:
Just a small question, will we ever need a `UInt64` for any of these as well as `UInt32`?

Reply:
argh.. turns out that `UInt32` extends `UInt64` (but not the other way). I wonder if this is a bug which we should fix, or if it makes sense in a weird way, because a `UInt32` also fits in 64 bits

Reply:
Ah thats right, currency is also supposed to be UInt64. I should probably set it to UInt64 as well, since it is also defined like this in all the Precondition interfaces